### DISCUSSION TITLE
How Reticulum deal with TCP/UDP/SCTP?

### DISCUSSION THREAD

**Discussion by hardBSDk** [markqvist/Reticulum#147] [Help & Questions] [answered]
Reticulum communication needs one of them? if it uses, which are the reason to choice that protocol?

https://www.geeksforgeeks.org/differences-between-tcp-and-udp/
https://www.geeksforgeeks.org/why-does-dns-use-udp-and-not-tcp/?ref=rp
https://www.geeksforgeeks.org/various-tcp-and-udp-ports/?ref=rp
https://www.geeksforgeeks.org/tcp-vs-udp-for-video-streaming/?ref=rp
https://www.geeksforgeeks.org/why-does-netflix-use-tcp-but-not-udp-for-streaming-video/?ref=rp
https://www.geeksforgeeks.org/why-is-youtube-using-tcp-but-not-udp/?ref=rp
https://www.geeksforgeeks.org/difference-between-sctp-and-tcp/
https://www.geeksforgeeks.org/difference-between-sctp-and-udp/
https://www.geeksforgeeks.org/sctp-full-form/
https://en.wikipedia.org/wiki/Stream_Control_Transmission_Protocol

**Comment by markqvist** [comment, marked as answer]
Reticulum does not implement a stream mode currently. This is intentional.

It is by no means impossible to implement something like this, but until an *very* well argued for use case for this comes along, I will not add it ;)

In practice, and specifically within the model that Reticulum operates under, almost all applications are better suited with a transactional model, in which the *receiver* of the data *requests* it. It is the receiver that is in control.

This turns the model of TCP/IP on it's head, where it is the sender that controls the data stream. This is more or less just a historic artifact of early protocol design, and it is not very useful. It also leads to *lots* of problems over slow links, and especially when there is a big assymetry in link capacity between sender and reciever. The model Reticulum uses solves these issues.
